src/Infrastructure/Translation/GreetingGenerator.php

```diff
@@ -4,87 +4,68 @@
+			'%spacedTitle%' => self::getSpacedTitle( $title ),
+			'%firstName%' => $firstName,
+			'%lastName%' => $lastName,
+		] );
 	}
 
 	public function createInformalGreeting( ?string $salutation, ?string $firstName, ?string $lastName ): string {
-		if ( ( $salutation !== self::GREETING_FAMILY && !$firstName ) ||
-			( $salutation === self::GREETING_FAMILY && !$lastName ) ) {
-			return $this->translator->trans( 'mail_introduction_generic' );
-		}
+		$salutationConfig = $this->getSalutationConfig( $salutation );
 
-		switch ( $salutation ) {
-			case self::GREETING_MALE:
-				return $this->translator->trans( 'mail_introduction_male_informal', [ '%firstName%' => $firstName ] );
-			case self::GREETING_FEMALE:
-				return $this->translator->trans( 'mail_introduction_female_informal', [ '%firstName%' => $firstName ] );
-			case self::GREETING_FAMILY:
-				return $this->translator->trans( 'mail_introduction_family_informal', [ '%lastName%' => $lastName ] );
-			default:
-				return $this->translator->trans( 'mail_introduction_generic' );
+		if ( !$firstName || !$lastName || !$salutationConfig ) {
+			return $this->translator->trans( $this->genericGreeting );
 		}
+
+		return $this->translator->trans( $salutationConfig['greetings']['informal'], [
+			'%firstName%' => $firstName,
+			'%lastName%' => $lastName,
+		] );
 	}
 
-	public function createInformalLastnameGreeting( ?string $salutation, ?string $lastName, ?string $title ): string {
-		if ( !$lastName ) {
-			return $this->translator->trans( 'mail_introduction_generic' );
-		}
+	public function createInformalLastnameGreeting( ?string $salutation, ?string $firstName, ?string $lastName, ?string $title ): string {
+		$salutationConfig = $this->getSalutationConfig( $salutation );
 
-		$spacedTitle = self::getSpacedTitle( $title );
-
-		switch ( $salutation ) {
-			case self::GREETING_MALE:
-				return $this->translator->trans( 'mail_introduction_male_lastname_informal', [
-					'%spacedTitle%' => $spacedTitle,
-					'%lastName%' => $lastName
-				] );
-			case self::GREETING_FEMALE:
-				return $this->translator->trans( 'mail_introduction_female_lastname_informal', [
-					'%spacedTitle%' => $spacedTitle,
-					'%lastName%' => $lastName
-				] );
-			case self::GREETING_FAMILY:
-				return $this->translator->trans( 'mail_introduction_family_informal', [
-					'%spacedTitle%' => $spacedTitle,
-					'%lastName%' => $lastName
-				] );
-			default:
-				return $this->translator->trans( 'mail_introduction_generic' );
+		if ( !$lastName || !$salutationConfig ) {
+			return $this->translator->trans( $this->genericGreeting );
 		}
+
+		return $this->translator->trans( $salutationConfig['greetings']['lastNameInformal'], [
+			'%spacedTitle%' => self::getSpacedTitle( $title ),
+			'%firstName%' => $firstName,
+			'%lastName%' => $lastName,
+		] );
+	}
+
+	private function getSalutationConfig( ?string $salutation ): ?array {
+		return $salutation ? $this->salutations->getSalutation( $salutation ) : null;
```

Comment:
I always try to avoid `null` values. The following method will also guarantee a result so you can get rid of the `|| $salutationConfig` checks and have calls like  `$this->translator->trans( $this->getSalutationTranslationKey( $salutation, 'formal' )`

```suggestion
	private function getSalutationTranslationKey( ?string $salutation, string $greetingType ): string {
	        if (!$salutation ) { return $this->genericGreeting; }
	        return $this->salutations->getSalutation( $salutation )['greetings'][$greetingType];
```

It also has the benefit of being more explicit what the return value is (a translation key) and masks the nested array structure.